commons/src/main/java/org/infinispan/commons/api/functional/FunctionalMap.java

```diff
@@ -0,0 +1,634 @@
+    *    these read/query operations, which sometimes can be expensive since
+    *    they involve talking to a remote node in the cluster or the
+    *    persistence layer, can be avoided optimising write-only operations.
+    *    </li>
+    *    <li>When ported over to the main Infinispan code base, write-only
+    *    maps would be expected to acquire locks on keys before executing
+    *    functions.</li>
+    * </ul>
+    */
+   interface WriteOnlyMap<K, V> extends FunctionalMap<K, V> {
+      /**
+       * Tweak write-only functional map executions providing {@link Param} instances.
+       */
+      WriteOnlyMap<K, V> withParams(Param<?>... ps);
+
+      /**
+       * Evaluate a write-only {@link BiConsumer} operation, with a value
+       * passed in and a {@link WriteEntryView} of the value associated with
+       * the key, and return a {@link CompletableFuture} which will be
+       * completed when the operation completes.
+       *
+       * By returning {@link CompletableFuture} instead of the function's
+       * return type directly, the method hints at the possibility that to
+       * executing the function might require remote data present in either
+       * a persistent store or a remote clustered node.
+       *
+       * This method can be used to implement single-key write-only operations
+       * which do not need to query previous value, such as:
+       *
+       * <ul>
+       * <li>{@link javax.cache.Cache#put(Object, Object)}</li>
+       * </ul>
+       *
+       * DESIGN RATIONALES:
+       * <ul>
+       *    <li>Since this is a write-only operation, no entry attributes can be
+       *    queried, hence the only reasonable thing can be returned is Void.
+       *    </li>
+       *    <li>Why provide this operation? Isn't {@link #eval(Object, Consumer)} enough?
+       *    The functionality provided by this function could indeed be implemented
+       *    with {@link #eval(Object, Consumer)}, but there's a crucial difference.
+       *    If you want to store a value and reference the value to be stored
+       *    from the passed in operation, {@link #eval(Object, Consumer)} needs
+       *    to capture that value. Capturing means that each time the operation
+       *    is called, a new lambda needs to be instantiated. By offering a
+       *    {@link BiConsumer} that takes user provided value as first parameter,
+       *    the operation does not capture any external objects when implementing
+       *    simple operations such as {@link javax.cache.Cache#put(Object, Object)},
+       *    and hence, the {@link BiConsumer} could be cached and reused each
+       *    time it's invoked.
+       *    </li>
+       * </ul>
+       *
+       * @param key the key associated with the {@link WriteEntryView} to be
+       *            passed to the operation
+       * @param value value to write, passed in as first parameter to the
+       *              {@link BiConsumer} operation.
```

Comment:
It depends on whether you can apply the delta without reading the previous value associated with the key. If you can apply it without needing to read, then yes. If not, no but you could use `ReadWriteMap.eval(K key, V value, BiFunction...`